Sources/Moya/Task+Internal.swift

```diff
@@ -0,0 +1,91 @@
+import Foundation
+import Moya
+
+extension Task {
+    typealias Parameters = (Encodable, ParameterEncoder)
```

Comment:
This might be a personal preference, but I'd rather have a proper struct here instead of relying on a tuple 😬 

Thoughts on using something like this instead? 
```swift
public struct Parameters {
      public let encodable: Encodable
      public let encoder: ParameterEncoder

      public init(encodable: Encodable, encoder: ParameterEncoder) {
          self.encodable = encodable
          self.encoder = encoder
      }
  }
```

I feel my main problem with tuples is the non-obvious usage, like

```swift
return try task.allParameters().reduce(request) { initialRequest, parameters in
  let encodable = parameters.0
  let encoder = parameters.1
```

Reply:
That indeed could be a possibility, I find structs more elegant than tuples too.
But I've chosen to use tuples internally to avoid creating an other struct and because it's only used from `Task` to `URLRequest`. On the other side, I use structs for the public API (`URLParams` and `CustomParams`).
Using structs on both sides could lead to reusing some code, i'll see where it brings me.

Reply:
I'm quite pleased with this change: `URLParams` and `CustomParams` now inherit from `Task.Parameters`.